mscore/palette.cpp

```diff
@@ -348,13 +354,127 @@ void Palette::mouseDoubleClickEvent(QMouseEvent* ev)
+                || element->type() == Element::Type::VBOX
+                || element->type() == Element::Type::HBOX
+                || element->type() == Element::Type::TBOX
+                || element->type() == Element::Type::MEASURE
+                || element->type() == Element::Type::BRACKET
+                || (element->type() == Element::Type::ICON
+                    && (static_cast<Icon*>(element)->iconType() == IconType::VFRAME
+                        || static_cast<Icon*>(element)->iconType() == IconType::HFRAME
+                        || static_cast<Icon*>(element)->iconType() == IconType::TFRAME
+                        || static_cast<Icon*>(element)->iconType() == IconType::MEASURE
+                        || static_cast<Icon*>(element)->iconType() == IconType::BRACKETS))) {
+                  Measure* last = sel.endSegment() ? sel.endSegment()->measure() : nullptr;
+                  for (Measure* m = sel.startSegment()->measure(); m; m = m->nextMeasureMM()) {
+                        QRectF r = m->staffabbox(sel.staffStart());
+                        QPointF pt(r.x() + r.width() * .5, r.y() + r.height() * .5);
+                        pt += m->system()->page()->pos();
+                        applyDrop(score, viewer, m, element, pt);
+                        if (m == last)
+                              break;
+                        }
+                  }
+            else if (element->type() == Element::Type::CLEF
+                     || element->type() == Element::Type::KEYSIG
+                     || element->type() == Element::Type::TIMESIG) {
+                  Measure* m1 = sel.startSegment()->measure();
+                  Measure* m2 = sel.endSegment() ? sel.endSegment()->measure() : nullptr;
+                  if (m2 == m1 && sel.startSegment()->rtick() == 0)
+                        m2 = nullptr;     // don't restore original if one full measure selected
+                  else if (m2)
+                        m2 = m2->nextMeasureMM();
+                  // for clefs, apply to each staff separately
+                  // otherwise just apply to top staff
+                  int staffIdx1 = sel.staffStart();
+                  int staffIdx2 = element->type() == Element::Type::CLEF ? sel.staffEnd() : staffIdx1 + 1;
+                  for (int i = staffIdx1; i < staffIdx2; ++i) {
+                        // for clefs, use mid-measure changes if appropriate
+                        Element* e1 = nullptr;
+                        Element* e2 = nullptr;
+                        // use mid-measure clef changes as appropriate
+                        if (element->type() == Element::Type::CLEF) {
+                              if (sel.startSegment()->segmentType() == Segment::Type::ChordRest && sel.startSegment()->rtick() != 0) {
+                                    ChordRest* cr = static_cast<ChordRest*>(sel.startSegment()->element(i * VOICES));
+                                    if (cr->isChord())
```

Comment:
Ooops, yes, seems it would crash with different rhythms on different staves and mid-measure clef changes.  I should probably find the next cr.  I'll make that change  and squah/rebase a bit later, also see if you find anything else that needs fixing.